File: Java/prog_lab_7/src/prog_lab_7/line.java
package prog_lab_7;

public class line {
	point ftpoint, sdpoint;
	
	public line() {
		ftpoint = new point();
		sdpoint = new point();
	}
	
	public void vvod() {
		this.ftpoint.vvod();
		this.sdpoint.vvod();
	}
	
	public void vivod()	{
		System.out.printf("(%f; %f;   %f; %f)\n", this.ftpoint.getx(), this.ftpoint.gety(), this.sdpoint.getx(), this.sdpoint.gety());
	}

	public double length() {
		return this.ftpoint.distance(this.sdpoint);
	}
}

File: C++/prog_lab_4/line.cpp
#define _CRT_SECURE_NO_WARNINGS
#include "line.h"
#include <iostream>

void line::vvod()
{
	ftpoint.vvod();
	sdpoint.vvod();
	return;
}

void line::vivod()
{
	printf("Координаты первой точки: x=%.3f y=%.3f\nКоординаты второй точки: x=%.3f y=%.3f",ftpoint.getx(), ftpoint.gety(), sdpoint.getx(), sdpoint.gety());
	return;
}

float line::length()
{
	return ftpoint.distance(&sdpoint);
}